core/classes/Widgets/ProfileWidgetBase.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+abstract class ProfileWidgetBase implements Widget {
```

Comment:
Profile widgets allow modules to create widgets that will be displayed on a user profile page.
The reactions one in this PR is an example of this system being used, but I envision it being used a lot more.

I want to add the ability to permission-gate these, so that site admins can control which profile widgets show on which profiles.

Some ideas:
- When we merge the friends/followers system
- Resources